iotfunctions/metadata.py

```diff
@@ -567,7 +567,57 @@ def set_params(self, **params):
         for key,value in list(params.items()):
             setattr(self, key, value)
         return self
-    
+
+
+    '''
+    Check if dataframe columns type is equivalent to the data item that is defined in the metadata
+    It checks the entire list of data items. Thus, depending where this code is executed, the dataframe might not be completed.
+    An exception is generated if there are not incompatible types of matching items AND and flag throw_error is set to TRUE
+    '''
+    def check_data_items_type(self, df, items, throw_error=False):
+
+        invalid_data_items = []
+
+        dict_types = {
+            'NUMBER' : ['int64','float64'],
+            'TIMESTAMP' : ['datetime64'],
+            'LITERAL' : ['O'],
+            'BOOLEAN' : ['bool']
+        }
+
+        if df is not None:
+            logger.info('Dataframe types:')
+            logger.info(df.dtypes)
+
+            for item in list(items.data_items):  #transform in list to iterate over it
+                df_column = {}
+                try:
+                    data_item = items.get(item)  #back to the original dict to retrieve item object
+                    df_column = df[data_item['name']]
+                except KeyError:
+                    logger.debug('Data item %s is not part of the dataframe yet.'% item)

```

Comment:
```python
logger.debug('Data item %s is not part of the dataframe yet.', item)  # ?
```

Reply:
If the function is invoked in the middle of the pipeline execution, some data items might not have been calculated yet, thus the dataframe won't contain the respective columns. This is only a warning, we can we remove the log if that is the case.

Reply:
@leucir 

```python
logger.debug('Data item %s is not part of the dataframe yet.'% item)
                                                             ^
logger.debug('Data item %s is not part of the dataframe yet.', item)  # ?
                                                             ^
```